src/DotNetty.Common/ThreadLocalPool.cs

```diff
@@ -273,9 +286,19 @@ internal bool Transfer(Stack dst)
                     DefaultHandle[] srcElems = head.elements;
                     DefaultHandle[] dstElems = dst.elements;
                     int newDstSize = dstSize;
+                    if (head.elements == null)
+                    {
+                        return false;
+                    }
+
                     for (int i = srcStart; i < srcEnd; i++)
                     {
                         DefaultHandle element = srcElems[i];
+                        if (element == null)
+                        {
+                            return false;
+                        }
+
```

Comment:
Question: shouldn't those elements be just ignored, instead of stopping everything